backend/testClient.spec.js

```diff
@@ -0,0 +1,175 @@
+const { createTestClient } = require('apollo-server-testing');
+
+const { typeDefs } = require("./typeDefs");
+const { resolvers } = require("./resolvers");
+
+const { ApolloServer, gql } = require('apollo-server');
+
+const { jwt } = require('jsonwebtoken');
+
+let token;
+
+describe("ServerTest", () => { 
+
+  const server = new ApolloServer({
+    typeDefs,
+    resolvers
+  });
+
+  const { query, mutate } = createTestClient(server);
+
+  it('test Login', async () => {
+    const res = await mutate({ mutation: LOGIN_USER,
+      variables: { username: "user1", password: "12345"}
+    });
+    token = res.data.loginUser.token;
+
+    let res2 = await query({ query: GET_TODOS, 
+      variables: {token: token}})
+  
+    expect(res2).toMatchObject({
```

Comment:
Make multiple test cases out of this:

* login
  * with valid credentials
    * responds with a JWT token in `data`.
  * with invalid credentials
    * responds with `errors` and `data` null
* todos
  * with a valid JWT
    * responds with todos
  * with invalid JWT
    * responds with `errors`